app/routes/application.js

```diff
@@ -28,13 +31,23 @@ export default TravisRoute.extend(BuildFaviconMixin, {
     });
 
     if (config.metricsAdapters.length > 0) {
-      const { metrics, router } = this;
+      const { metrics, router, controller } = this;
       router.on('routeDidChange', () => {
         try {
           const { currentURL: page } = router;
-          metrics.trackPage({ page });
-        } catch (err) {
-        }
+          let needsReset = true;
+
+          const hitCallback = () => {
+            needsReset = false;
+            controller && controller.resetUTMs();
+          };
+
+          metrics.trackPage({ page, hitCallback });
+
+          later(() => {
+            if (needsReset) this.controller.resetUTMs();
+          }, RESET_UTM_PARAMS_DELAY);
```

Comment:
I'm not sure if it makes a difference here, but `later` was a part of the test slowdown that I recently looked into. In that case, it seemed that awaiting `settled` or `visit` resulted in trying to wait for the `later` call to clear. But I also noticed that you put an explicit timeout after settled in one of your tests here, so maybe it's not happening in this case.

If it does seem like it may be an issue in this case, we could define the delay as an `interval` in `config/environment.js`, and then set a lower number (or 0) in the test environment, as has been done for a few other delays/intervals.

Reply:
That's a great point! Thank you @sorta, I'll add an interval to the config and set it to `0` for tests 👍